doc/data_io_pipeline_design.md

```diff
@@ -0,0 +1,20 @@
+# Data IO Pipeline Design
+
+Data IO pipeline for ElasticDL involves reading data from [RecordIO](https://github.com/elasticdl/recordio) file or ODPS table, making data generator for tf.data.Dataset and parsing features and labels by dataset_fn user defined (see <em> Figure 1</em>).
+
```

Comment:
It seems that we have an assumption here, which data has been pre-processed already.

The full data IO pipeline should be:

```
original data ---> data preprocess ---> feature engineering ---> transformed to tf.dataset . --> user parsing and provide to model
```

Reply:
Data can be pre-processed in dataset_fn, or pre-processed before DataReader.

Reply:
Yes, feature engineering part is not implemented yet. We plan to add "feature engineering --->" node after finish implementation.